Added cache invalidation to preview queries

ronaldlangeveld opened the pull request:
ref https://linear.app/ghost/issue/PLG-384/paid-only-content-not-showing-in-web-preview-for-paid-member

Previously, `handleCacheInvalidation` only purged the base preview path (`/p/<UUID>/`) for draft and scheduled posts, leaving variants with query params (e.g., `?member_status=free`) cached in Fastly/Varnish. This caused stale content to persist in routes with the query parameters.

This change updates:
- `handleCacheInvalidation` to return a comma-separated list of URLs—including the base path and variants with `member_status=anonymous`, `free`, and `paid`—when a draft post changes (and wasn’t previously published) or a scheduled post is modified.
- `getCacheHeaderFromEventString` in the controller to apply the same logic for `scheduled_updated` and `draft_updated` events.

This change updates `handleCacheInvalidation` to return a comma-separated list of preview paths and variants with `member_status=anonymous`, `free`, and `paid` when a draft post changes or a scheduled post is modified.

ronaldlangeveld committed:
Added cache invalidation to preview queries

ref https://linear.app/ghost/issue/PLG-384/paid-only-content-not-showing-in-web-preview-for-paid-member

Previously, `handleCacheInvalidation` only purged the base preview path
(`/p/<UUID>/`) for draft and scheduled posts, leaving variants with
query params (e.g., `?member_status=free`) cached in Fastly/Varnish.
This caused stale content to persist in routes with the query
parameters.

This change updates `handleCacheInvalidation` to return a comma-separated list
of preview paths and variants with `member_status=anonymous`, `free`, and `paid`
when a draft post changes or a scheduled post is modified.

ronaldlangeveld committed:
Updated posts controller

Ghost-Slimer commented:
Deployed to **staging** with ID: `3312`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

ronaldlangeveld committed:
Added space between url to invalidate

- added a space after each path to invalidate in the string to align
  with other areas where we use multiple urls in a string

Ghost-Slimer commented:
Deployed to **staging** with ID: `3312`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

Ghost-Slimer commented:
Deployed to **staging** with ID: `3312`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>
